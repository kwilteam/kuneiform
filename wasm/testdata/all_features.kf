database td1;
use a_ext{addr: '0x0000', seed: 3} as ext1;

table tt1 { tc1 int }

action act1($var1) private {
    select * from tt1 where tc1 = $var1;
}

init() {
    act1(1);
    $v = ext1.call(@caller);
    insert into tt1 values ($v);
}