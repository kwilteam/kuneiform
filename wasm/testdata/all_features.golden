{
  "owner": "",
  "name": "td1",
  "tables": [
    {
      "name": "tt1",
      "columns": [
        {
          "name": "tc1",
          "type": "int"
        }
      ]
    }
  ],
  "actions": [
    {
      "name": "act1",
      "inputs": [
        "$var1"
      ],
      "public": false,
      "mutability": "update",
      "auxiliaries": null,
      "statements": [
        "select * from tt1 where tc1 = $var1;"
      ]
    },
    {
      "name": "init",
      "inputs": null,
      "public": false,
      "mutability": "update",
      "auxiliaries": null,
      "statements": [
        "act1(1);",
        "$v=ext1.call(@caller);",
        "insert into tt1 values ($v);"
      ]
    }
  ],
  "extensions": [
    {
      "name": "a_ext",
      "config": [
        {
          "argument": "addr",
          "value": "'0x0000'"
        },
        {
          "argument": "seed",
          "value": "3"
        }
      ],
      "alias": "ext1"
    }
  ]
}